qlib/__init__.py

```diff
@@ -20,11 +20,17 @@ def init(default_conf="client", **kwargs):
     from .config import C
     from .data.cache import H
 
-    H.clear()
-
     # FIXME: this logger ignored the level in config
     logger = get_module_logger("Initialization", level=logging.INFO)
 
+    skip_if_reg = kwargs.pop("skip_if_reg", False)
+    if skip_if_reg and C.registered:
```

Comment:
Should we also check if `R` contains active experiment here? It seems that only the function `auto_init` has been enabled skipping initialization for now.